mne/simulation/source.py

```diff
@@ -267,6 +267,13 @@ def simulate_stc(src, labels, stc_data, tmin, tstep, value_fun=None,
         hemi_ind = hemi_to_ind[label.hemi]
         src_sel = np.intersect1d(src[hemi_ind]['vertno'],
                                  label.vertices)
+        if len(src_sel) == 0:
+            idx = np.searchsorted(src[hemi_ind]['vertno'],
+                                  label.vertices)
+            src_len = len(src[hemi_ind]['vertno']) - 1
+            idx[idx > src_len] = src_len
+            src_sel = np.unique(src[hemi_ind]['vertno'][idx])
```

Comment:
@makkostya by writing this do you assume that if you get an idx that is close to label.vertices that it is physically close to the label on the cortex? if so this is likely broken as you cannot guarantee this. Either you need to grow the label of I would suggest to indeed write a meaningful error message in an exception

following an IRL discussion with @massich

Reply:
@agramfort And what if select the vertex from the source space which is the closest in terms of Euclidean norm?

Reply:
To me that would make sense. 
To sum up the problem, if I understand properly, we have 2 sets of identifiers (which they all have position somewhere else). Let's assume that those sets are number (id) and their position is (id, 0, 0). Then 1 is colser to 2 than to 3 and so on.
(for sake of abstraction I'll call src[hemi_ind]['vertno']-> foo, and  label.vertices->bar)

case A: there is the intersection of the identifiers
```py
    foo=[3, 6, 9]
    bar=[0, 1, 2, 3, 4, 5, 6]  
    result = my_func(foo, bar) # [3, 6]
```

case B: there is no intersection, return set(the closest foo id of the bar element)
```py
    foo=[3, 6, 9]
    bar=[0, 1, 2, 3, 4, 5, 6]  
    result = my_func(foo, bar) # [3, 3, 3, 3, 3, 6, 5]
    resut = set(result) # [3, 6]
```

Then would happen in someting like:
```py
my_func(foo=[3, 6, 9], bar=[0, 1, 2, 3, 4, 5, 7, 8]) # [3]
my_func(foo=[3, 6, 9], bar=[0, 1, 2, 4, 5, 7, 8]) # [3, 6, 9]
```

Question, do we care about the size of the resulting set? is it better `[3]` than `[3, 6, 9]`?

Maybe we should always take the set of the closest element (with a threshold if we consider that the hull of two sets might no intersect)